gcc/rust/backend/rust-compile-expr.cc

```diff
@@ -958,12 +958,71 @@ CompileExpr::visit (HIR::LiteralExpr &expr)
     }
 }
 
+bool
+lvalue_p (const_tree ref)
```

Comment:
this should be renamed and moved into rust-compile-base.cc or into rust-tree.cc call it is_lvalue_tree